recipes/flatcc/all/conanfile.py

```diff
@@ -0,0 +1,117 @@
+import os
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+class FlatccConan(ConanFile):
+    name = "flatcc"
+    description = "C language binding for Flatbuffers, an efficient cross platform serialization library"
+    topics = "conan", "flatbuffers", "serialization"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/dvidelabs/flatcc"
+    license = "Apache-2.0"
+    options = { "shared": [True, False],
+                "fPIC": [True, False],
+                "portable": [True, False],
+                "gnu_posix_memalign": [True, False],
+                "runtime_lib_only": [True, False],
+                "verify_assert": [True, False],
+                "verify_trace": [True, False],
+                "reflection": [True, False],
+                "native_optim": [True, False],
+                "fast_double": [True, False],
+                "ignore_const_condition": [True, False],
+    }
+    default_options = { "shared": False,
+                        "fPIC": True,
+                        "portable": False,
+                        "gnu_posix_memalign": True,
+                        "runtime_lib_only": False,
+                        "verify_assert": False,
+                        "verify_trace": False,
+                        "reflection": True,
+                        "native_optim": False,
+                        "fast_double": False,
+                        "ignore_const_condition": False
+    }
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "cmake"
+    exports_sources = ["CMakeLists.txt"]
+
+    _cmake = None
+
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.os == "Windows":
+            if self.settings.compiler == "Visual Studio" and self.options.shared:
+                #Building flatcc shared libs with Visual Studio is broken
+                raise ConanInvalidConfiguration("Building flatcc libraries shared is not supported")
+            if self.settings.compiler == "gcc":
+                raise ConanInvalidConfiguration("Building flatcc with MinGW is not supported")
+        if self.settings.os == "Macos" and self.options.shared:
+            raise ConanInvalidConfiguration("Building flatcc shared on Macos is currently not supported due to Systems Integrity Protection problems")
```

Comment:
The package builds, the problem arises when using it from inside other applications.... but some users will run it directly or from the Conan `build()` method. I don't think we should forbid this configuration.

```suggestion
```

I agree this is a problem if you want to test this package using `add_custom_command(... COMMAND flatcc ...)` and I'm really concern about this issue (I'm investigating some alternatives here https://github.com/conan-io/conan/pull/7324). Meanwhile, there is a different alternative that works: calling `flatcc` from the `build()` method in the `test_package/conanfile.py` file:

```
def build(self):
    # Generate the message using flatcc
    self.run("flatcc ....", run_environment=True)

    # Compile the library
    cmake = CMake(self)
    cmake.configure()
    cmake.build()
```

Reply:
In my opinion running flatcc via a custom command in cmake is the proper way to generate the sources. This way the C source files will be automatically re-generated when the flatbuffer schema changes. I created an additional cmake [module](https://github.com/wdobbe/cmake_extra/blob/master/modules/FlatccGenerator.cmake) to make this easy.

Calling flatcc from the build() function will work if the package is built as a dependency, but not (I think) if you want to develop the package itself. For example if you run `conan install` for a conan package that uses flatcc, Conan will install flatcc, but the flatbuffer C files are not generated. So trying to build the project will fail.

Reply:
Yes, I totally understand your concerns, but having this `raise ConanInvalidConfiguration` in the graph for `Macos-shared` will invalidate any graph for a build-profile like:

```ini
[settings]
os=Macos
...
[options]
*:shared=True
```

Conan will evaluate the graph and raise. Not everyone will use `flatcc` the same way, some people might be using it in the `build()` function, others might want to create a virtualenv with it in the path,... and we are blocking them all with this exception. 

Some of them, those who want to use `flatcc` from inside another application in Mac, can apply a workaround and add `flatcc:shared=False` if needed.